src/main/java/picard/analysis/CollectWgsMetrics.java

```diff
@@ -583,69 +606,78 @@ protected WgsMetricsCollector getCollector(final int coverageCap, final Interval
 
                 pileupSize++;
-                if (pileupSize <= coverageCap) {
-                    baseQHistogramArray[recs.getRecord().getBaseQualities()[recs.getOffset()]]++;
-                }
             }
 
-            final int depth = Math.min(pileupSize, coverageCap);
-            if (depth < pileupSize) basesExcludedByCapping += pileupSize - coverageCap;
-            depthHistogramArray[depth]++;
+            final int highQualityDepth = Math.min(pileupSize, coverageCap);
+            if (highQualityDepth < pileupSize) basesExcludedByCapping += pileupSize - coverageCap;
+            highQualityDepthHistogramArray[highQualityDepth]++;
+            unfilteredDepthHistogramArray[unfilteredDepth]++;
+
+            // check that we added the same number of bases to the raw coverage histogram and the base quality histograms
+            if (Arrays.stream(unfilteredBaseQHistogramArray).sum() !=  LongStream.rangeClosed(0,coverageCap).map(i -> (i * unfilteredDepthHistogramArray[(int)i])).sum()) {
+                throw new PicardException("updated coverage and baseQ distributions unequally");
+            }
         }
 
         public void addToMetricsFile(final MetricsFile<WgsMetrics, Integer> file,
                                      final boolean includeBQHistogram,
                                      final CountingFilter dupeFilter,
                                      final CountingFilter mapqFilter,
                                      final CountingPairedFilter pairFilter) {
-            addMetricsToFile(file, dupeFilter, mapqFilter, pairFilter);
+            final WgsMetrics metrics = getMetrics(dupeFilter, mapqFilter, pairFilter);
+
+            // add them to the file
+            file.addMetric(metrics);
+            file.addHistogram(getHighQualityDepthHistogram());
 
             if (includeBQHistogram) {
                 addBaseQHistogram(file);
             }
         }
 
         protected void addBaseQHistogram(final MetricsFile<WgsMetrics, Integer> file) {
-            file.addHistogram(getBaseQHistogram());
+            file.addHistogram(getUnfilteredBaseQHistogram());
         }
 
-        protected void addMetricsToFile(final MetricsFile<WgsMetrics, Integer> file,
-                                        final CountingFilter dupeFilter,
-                                        final CountingFilter mapqFilter,
-                                        final CountingPairedFilter pairFilter) {
-            // get the depth histogram and metrics
-            final Histogram<Integer> depthHistogram = getDepthHistogram();
-            final WgsMetrics metrics = getMetrics(depthHistogram, dupeFilter, mapqFilter, pairFilter);
-
-            // add them to the file
-            file.addMetric(metrics);
-            file.addHistogram(depthHistogram);
+        protected Histogram<Integer> getHighQualityDepthHistogram() {
+            return getHistogram(highQualityDepthHistogramArray,"coverage", "count");
```

Comment:
if you do include multiple depth histograms the x axis is recycled, which means that you have to put enough information in the count label to make it clear what it is. I'd use "HighQualityDepthCount" etc.